### PR TITLE
Move serde with context

### PR DESCRIPTION
**Pull request by gz**
Is this a user-visible change (yes/no): no

### CODE REVIEW

File: crates/pipeline-types/Cargo.toml [hunk 1]
```diff
@@ -22,3 +22,6 @@ log = "0.4.20"
 utoipa = "4.1"
 proptest = { version = "1.0.0", optional = true }
 proptest-derive = { version = "0.3.0", optional = true }
+# Revert after https://github.com/paupino/rust-decimal/pull/637 is merged:
+rust_decimal = { git = "https://github.com/gz/rust-decimal.git", rev = "ea85fdf" }
+serde_arrow = { git = "https://github.com/gz/serde_arrow.git", features = ["arrow-50"], rev = "7b604f0" }
```
Comment (mihaibudiu) [inline crates/pipeline-types/Cargo.toml:27]: does this also depend on a specific PR?

File: crates/pipeline-types/src/serde_with_context/deserialize.rs [hunk 2]
```diff
@@ -548,295 +524,3 @@ macro_rules! deserialize_table_record {
         }
     }
 }
-
-#[cfg(test)]
```
Comment (mihaibudiu) [inline crates/pipeline-types/src/serde_with_context/deserialize.rs:552]: so these moved somewhere else?

File: crates/pipeline-types/src/serde_with_context/serialize.rs [hunk 3]
```diff
@@ -313,178 +307,17 @@ macro_rules! serialize_table_record {
...
-
-serialize_struct!(NeighborhoodDescr(K, V)[4]{
-    anchor["anchor"]: Option<K>,
-    anchor_val["anchor_val"]: V,
-    before["before"]: usize,
-    after["after"]: usize
-});
-
-#[cfg(test)]
-mod test {
```
Comment (mihaibudiu) [inline crates/pipeline-types/src/serde_with_context/serialize.rs:339]: these too?

**Review by mihaibudiu — APPROVED** [review]
Many tests were deleted without seemingly going somewhere else.
Is that a net loss?

File: crates/pipeline-types/src/serde_with_context/deserialize.rs (hunk 2 above)
Comment (gz) [reply crates/pipeline-types/src/serde_with_context/deserialize.rs:552]: um I hoped so, let me check

File: crates/pipeline-types/Cargo.toml (hunk 1 above)
Comment (gz) [reply crates/pipeline-types/Cargo.toml:27]: yes will add https://github.com/chmp/serde_arrow/pull/147

**Pull request merged**